create_patched_sdk.sh

```diff
@@ -1,182 +1,233 @@
-    xcode_sdk_real=$(realpath "$xcode_sdk_path")
-
-    if [[ $xcode_sdk_real == $xcode_default_sdk_path ]]; then
-        preferred_xcode_sdk_path=$xcode_sdk_path
-    fi
+	if [[ $xcode_sdk_real == $xcode_default_sdk_path ]]; then
+		preferred_xcode_sdk_path=$xcode_sdk_path
+	fi
 done
 
 if [[ -z $preferred_xcode_sdk_path ]]; then
-    printf 'Failed to find sdk for simulator runtime\n'
-    exit 1
+	echo 'Failed to find sdk for simulator runtime'
+	exit 1
 fi
 
 preferred_xcode_sdk_name=$(basename $preferred_xcode_sdk_path)
 
-xcode_sdk_ios_version=${preferred_xcode_sdk_name:8} # Remove 'iPhoneOS' in front of sdk name
-xcode_sdk_ios_version=${xcode_sdk_ios_version%????} # Remove '.sdk' at back of sdk name
+if [[ $sdk_platform == iOS ]]; then
+	xcode_sdk_version=${preferred_xcode_sdk_name:8} # Remove 'iPhoneOS' in front of sdk name
+else
+	xcode_sdk_version=${preferred_xcode_sdk_name:9} # Remove 'AppleTVOS' in front of sdk name
+fi
+xcode_sdk_version=${xcode_sdk_version%????} # Remove '.sdk' at back of sdk name
 
 sdks_output_path_single_sdk_path=""
 
-device_support_dir="$HOME/Library/Developer/Xcode/iOS DeviceSupport/"
+device_support_dir="$HOME/Library/Developer/Xcode/$sdk_platform DeviceSupport/"
 if [[ -d $device_support_dir ]] && ignored $use_simulator; then
-    for symbols_path in "$device_support_dir"*; do
-        if ! [[ -d $symbols_path ]]; then
-            continue
-        fi
-
-        ios_version="$(basename "$symbols_path" | grep -o "\d\+\(\.\d\+\)\{1,2\}")"
-        sdk_name=$(printf "iPhoneOS%s.sdk" $ios_version)
-
-        symbols_actual_path="$symbols_path/Symbols/System"
-        if ! [[ -d $symbols_actual_path ]]; then
-            printf "Symbols for iOS %s don't exist\n" "$ios_version"
-            continue
-        fi
-
-        sdks_output_path_single_sdk_path="$sdks_output_path/$sdk_name"
-        if [[ -d $sdks_output_path_single_sdk_path ]]; then
-            printf 'SDK for iOS %s already exists\n' "$ios_version"
-            continue
-        fi
-
-        printf 'Creating SDK for iOS %s ...\n' "$ios_version"
-
-        if [[ $xcode_sdk_ios_version != $ios_version ]]; then
-            printf "Warning: Xcode SDK for iOS %s will be used as a base for sdk for iOS %s\n" $xcode_sdk_ios_version "$ios_version"
-        fi
-
-        mkdir -p "$sdks_output_path_single_sdk_path"
-        cp -R "$xcode_default_sdk_path/"* "$sdks_output_path_single_sdk_path"
-
-        "$tbd_tool" \
-            -p $no_warnings "${tbd_options[@]}" "${archs_option[@]}" -v $version -r all "$symbols_actual_path" \
-            -o "${write_options[@]}" $no_overwrite "$sdks_output_path_single_sdk_path/System"
-
-        if [[ $? -ne 0 ]]; then
-            printf 'Failed to create tbds from Symbols directory for iOS %s\n' $ios_version
-        fi
-    done
+	for symbols_path in "$device_support_dir"*; do
+		if ! [[ -d $symbols_path ]]; then
+			continue
+		fi
+
+		version="$(basename "$symbols_path" | grep -o "\d\+\(\.\d\+\)\{1,2\}")"
```

Comment:
this overrides the `version` variable above (which is used for `tbd` version).

maybe `os_version`? and we can rename the other `version` variable as well for clarity; `tbd_version`